### PR TITLE
Add velo-project and deploy-folder as inputs

### PR DESCRIPTION
**Pull request by RmStorm**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -7,6 +7,14 @@ inputs:
   google_cloud_storage_bucket_name:
     description: "Name of the Google cloud storage bucket to upload the artifacts to"
     required: true
+  velo_project:
+    description: "Name of the Google cloud storage bucket to upload the artifacts to"
+    default: ${{ github.event.repository.name }}
```
Comment (ljodal) [inline action.yml:12]: Does this actually work? I suspect you might not be able to have expressions here 🤔

File: action.yml [hunk 2]
```diff
@@ -20,4 +28,4 @@ runs:
       uses: google-github-actions/setup-gcloud@v0
     - name: Upload deploy artifacts to bucket
       shell: bash
-      run: gsutil cp -r .deploy/* "gs://${{ inputs.google_cloud_storage_bucket_name }}/${{ github.event.repository.name }}/${{ github.sha }}/"
+      run: gsutil cp -r ${{ inputs.deploy_folder }}/* "gs://${{ inputs.google_cloud_storage_bucket_name }}/${{ inputs.  velo_project }}/${{ github.sha }}/"
```
Comment (ljodal) [inline action.yml:31]:
```suggestion
      run: gsutil cp -r ${{ inputs.deploy_folder }}/* "gs://${{ inputs.google_cloud_storage_bucket_name }}/${{ inputs.velo_project }}/${{ github.sha }}/"
```

File: action.yml (hunk 1 above)
Comment (RmStorm) [reply action.yml:12]: idk, any easy way to try?

Comment (ljodal) [reply action.yml:12]: Nope 🙃

Comment (RmStorm) [reply action.yml:12]: Can I just deploy it and see if it works :see_no_evil: ?

Comment (thor) [reply action.yml:12]: I don't think it works; if it doesn't, you can fallback to using shell script to use `github.event.repository.name` as the default if the parameter's unset.

Can't you use the action by `kolonialno/upload-velo-artifacts@roald/add-velo-project-and-deploy-folder`? 🙃

File: action.yml [hunk 3]
```diff
@@ -20,4 +28,4 @@ runs:
       uses: google-github-actions/setup-gcloud@v0
     - name: Upload deploy artifacts to bucket
       shell: bash
-      run: gsutil cp -r .deploy/* "gs://${{ inputs.google_cloud_storage_bucket_name }}/${{ github.event.repository.name }}/${{ github.sha }}/"
+      run: gsutil cp -r ${{ inputs.deploy_folder }}/* "gs://${{ inputs.google_cloud_storage_bucket_name }}/${{ inputs.velo_project }}/${{ github.sha }}/"
```
Comment (thor) [inline action.yml:31]:
```suggestion
      run: |
        USER_VELO_PROJECT="${{ inputs.velo_project }}"
        VELO_PROJECT="${USER_VELO_PROJECT:-"${{ github.event.repository.name }}"}"
        gsutil cp -r ${{ inputs.deploy_folder }}/* "gs://${{ inputs.google_cloud_storage_bucket_name }}/${VELO_PROJECT}/${{ github.sha }}/"
```
suggestion: re: previous thread

---

sidenote: this is how the diff looks to me before submitting it
image

File: action.yml (hunk 1 above)
Comment (ljodal) [reply action.yml:12]: Yeah, you can test it like that, but I also highly doubt it works. If it doesn't you can try something like this where it's used:

```bash
${INPUT_velo_project:-"${{ github.event.repository.name}}"}
```

**Review by thor — APPROVED** [review]
Without a close look, this looks good to me. I added a suggestion to try if the expression is a no-op.

File: action.yml (hunk 3 above)
Comment (ljodal) [reply action.yml:31]: Yeah, the GitHub rendering is borked 🙃 

Also note that the inputs already should be available as env variables: `INPUT_<variable name>`, so e.g. `INPUT_velo_project`

**Comment by ljodal** [comment]
I've not done anything to set up a publishing pipeline for this, so I think everyone using this for now is just using `@main`

**Pull request merged**